server/pulp/server/webservices/views/tasks.py

```diff
@@ -67,6 +69,32 @@ def get(self, request):
         serialized_task_statuses = [task_serializer(task) for task in raw_tasks]
         return generate_json_response_with_pulp_encoder(serialized_task_statuses)
 
+    @auth_required(authorization.DELETE)
+    def delete(self, request):
+        """
+        Delete the tasks with status as finished, error, timed-out and skipped
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+
+        :return: Response containing None or pulp Exception
+        :rtype:  django.http.HttpResponse or pulp.Exception
+        """
+        task_state = request.GET.getlist('state')
+        if not task_state:
+            raise pulp_exceptions.PulpCodedForbiddenException(
+                error_message=_("No task state given to parameters list for delete"))
+
+        for state in task_state:
+            if state not in VALID_STATES:
+                raise pulp_exceptions.PulpCodedForbiddenException(
+                    error_message=_("Invalid task state given as parameter for delete"))
+
+        for state in task_state:
+                TaskStatus.objects(state=state).delete()
+
+        return generate_json_response(None)
```

Comment:
I think this will produce a json document with the value ``null`` and put that in the response body, with a code of 200. That's a behavior seen in other parts of our API, but I don't think it's a good one. As noted in a previous comment, it would be better to return a 204 with an empty body.